wiki/TypeScript-Migration.md

```diff
@@ -0,0 +1,48 @@
+# TypeScript migration
+
+The following steps are recommended for migrating components to TypeScript.
+
+1. [Pick a component](https://hashicorp.atlassian.net/browse/HDS-2392), assign yourself to the task in Jira, and add it to the current sprint
+
+    - [This spreadsheet](https://docs.google.com/spreadsheets/d/1sWzLSP8TUb3WYibYNZ5w6UfEb338DQbZkVaTjK8b1tw/edit#gid=0) contains (more or less) all the dependencies between components, so it’s easier to understand what is ready to be converted (e.g. has dependencies that are already converted to TypeSscript) and what is not.
+
+2. Rename the backing class from `.js` to `.ts`; commit after this step to make it easier to review the changes
+
+3. Opening the files (`.ts` and `.hbs`) in VS Code should show you the linting errors (if you have the [Glint extension](https://marketplace.visualstudio.com/items?itemName=typed-ember.glint-vscode)); similarly, if you are running `yarn lint` in `packages/components`
+
+4. Define component signature
+
+    - You can either do it manually [following the documentation](https://typed-ember.gitbook.io/glint/environments/ember/component-signatures#glimmer-components)
+    - Or, you can generate it using `npx ember-codemod-args-to-signature --src /path/to/component`. This codemod renames the component class name to `IndexComponent` by default, so make sure to revert it (e.g. `HdsBadgeComponent`)
+
+5. Add types and use them in the signature
+
+    - The signature should follow the naming convention (e.g. `HdsBadgeSignature`) and should be placed right before the backing class declaration, so it’s consistent across components
+
+    - Compare the signature generated, with the expected arguments using the website documentation (Component API section), and start defining the [basic types](https://www.typescriptlang.org/docs/handbook/basic-types.html) (e.g. `boolean`, `number`, `string`) and mark them as optional using the `argument?: type` syntax.
+
```

Comment:
As discussed [here](https://github.com/hashicorp/design-system/pull/2108#issuecomment-2117306919). Feel free to edit as you see fit.

```suggestion

    - Try to keep the order of the arguments consistent between the `Args` in the signature and the "Component API" documentation, so it's easier to reference and compare one to another; there are no specific rules about the order, so look at other similar components as reference (for example, `Button`, `Badge/BadgeCount`, `IconTile`, `LinkInline/Standalone`, `Tag` they all share some common props, like `type`, `size`, `color`, `text`, `textSomething`, `icon`, `iconSomething` so try to keep this order; for component that get their arguments from the `Interactive` component, keep all the `Interactive`-related arguments in the same block and order; etc.)

```